src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -806,28 +828,220 @@ public async Task Delete()
                     {
                         // we only send the successful index here, we might fail to delete the index
                         // because a node is down, and we don't want to cause the client to wait on an
                         // index that doesn't exists in the Raft log
-                        [nameof(DeleteDatabaseResult.RaftCommandIndex)] = actualDeletionIndex,
+                        [nameof(DeleteDatabaseResult.RaftCommandIndex)] = lastDeletionIndex,
                         [nameof(DeleteDatabaseResult.PendingDeletes)] = new DynamicJsonArray(pendingDeletes)
                     });
                 }
             }
         }
 
+        private async Task WaitForRemovingNodesFromDatabases(
+            Dictionary<string, DeleteDatabaseDetails> databasesToDelete,
+            Dictionary<string, string> nodesTagsAndUrls, //clusterTopology.AllNodes
+            long lastDeletionIndex)
+        {
+            if (databasesToDelete.Count == 0)
+                return;
+
+            await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, lastDeletionIndex);
+            var timeout = Server.Configuration.Cluster.OperationTimeout.AsTimeSpan/databasesToDelete.Count;
+
+            var reqExecutors = new Dictionary<string, ClusterRequestExecutor>();
+            try
+            {
+                foreach (var (dbName, details) in databasesToDelete)
+                {
+                    var deleteCmdIndex = details.RaftIndex;
+                    var databaseNodes = details.Topology.AllNodes.ToHashSet();
+
+                    foreach (var removedNodeTag in details.NodesToRemove) // nodeTags
+                    {
+                        var removeNodeFromDbCmdGuid = $"RemoveNodeFromDatabase/{removedNodeTag}/{deleteCmdIndex}";
+                        var removeNodeFromDbCmdRaftIndex = 0L;
+                        var sw = Stopwatch.StartNew();
+                        while (TryGetLastCommittedDeleteRaftIndexByGuid(ServerStore, removeNodeFromDbCmdGuid, out removeNodeFromDbCmdRaftIndex) == false)
+                        {
+                            if (sw.Elapsed > timeout)
+                            {
+                                throw new TimeoutException(
+                                    $"RemoveNodeFromDatabase command of node {removedNodeTag} doesn't exist in the log of node {ServerStore.NodeTag}.");
+                            }
+
+                            await Task.Delay(TimeSpan.FromMilliseconds(300));
+                        }
+
+                        using var cts = new CancellationTokenSource(timeout - sw.Elapsed);
+                        await WaitForDatabaseNotificationOnCluster(dbName, removeNodeFromDbCmdRaftIndex, nodesTagsAndUrls, databaseNodes, reqExecutors, cts.Token);
+                    }
+                }
+            }
+            finally
+            {
+                foreach (var requestExecutor in reqExecutors.Values)
```

Comment:
let's dispose the requestExecutor at the same place we create it

Reply:
You asked for it to be like that in here:
https://github.com/ravendb/ravendb/pull/16272#discussion_r1252274418